CameraXBasic/app/src/main/java/com/android/example/cameraxbasic/fragments/CameraFragment.kt

```diff
@@ -277,60 +281,73 @@ class CameraFragment : Fragment() {
-        }.build()
-
-        imageAnalyzer = ImageAnalysis(analyzerConfig).apply {
-            setAnalyzer(mainExecutor,
-                    LuminosityAnalyzer { luma ->
+
+        val rotation = viewFinder.display.rotation
+
+        // Bind the CameraProvider to the LifeCycleOwner
+        val cameraSelector = CameraSelector.Builder().requireLensFacing(lensFacing).build()
+        val cameraProviderFuture = ProcessCameraProvider.getInstance(requireContext())
+        cameraProviderFuture.addListener(Runnable {
+
+            // CameraProvider
+            val cameraProvider: ProcessCameraProvider = cameraProviderFuture.get()
+
+            // Preview
+            preview = Preview.Builder()
+                // We request aspect ratio but no resolution
+                .setTargetAspectRatio(screenAspectRatio)
+                // Set initial target rotation
+                .setTargetRotation(rotation)
+                .build()
+
+            // Default PreviewSurfaceProvider
+            preview?.setPreviewSurfaceProvider(viewFinder.previewSurfaceProvider)
+
+            // ImageCapture
+            imageCapture = ImageCapture.Builder()
+                .setCaptureMode(ImageCapture.CAPTURE_MODE_MINIMIZE_LATENCY)
+                // We request aspect ratio but no resolution to match preview config, but letting
+                // CameraX optimize for whatever specific resolution best fits requested capture mode
+                .setTargetAspectRatio(screenAspectRatio)
+                // Set initial target rotation, we will have to call this again if rotation changes
+                // during the lifecycle of this use case
+                .setTargetRotation(rotation)
+                .build()
+
+            // ImageAnalysis
+            imageAnalyzer = ImageAnalysis.Builder()
+                // We request aspect ratio but no resolution
+                .setTargetAspectRatio(screenAspectRatio)
+                // Set initial target rotation, we will have to call this again if rotation changes
+                // during the lifecycle of this use case
+                .setTargetRotation(rotation)
+                .build()
+                // The analyzer can then be assigned to the instance
+                .also {
+                    it.setAnalyzer(mainExecutor, LuminosityAnalyzer {luma ->
                         // Values returned from our analyzer are passed to the attached listener
-                        // We log image analysis results here --
-                        // you should do something useful instead!
-                        val fps = (analyzer as LuminosityAnalyzer).framesPerSecond
-                        Log.d(TAG, "Average luminosity: $luma. " +
-                                "Frames per second: ${"%.01f".format(fps)}")
+                        // We log image analysis results here - you should do something useful instead!
+                        Log.d(TAG, "Average luminosity: $luma")
```

Comment:
Have closed the previous PR because of too many reverted commits, in order to provide a better overview and not to have all of them in the commit log. The reason why I removed the FPS is because it only returned value `infinite` and not any useful numeric value. If I'd revert this, I'd have to set the analyzer in two steps again, so that I could get a handle.